tests/e2e/upgrades_test.go

```diff
@@ -22,27 +22,29 @@ import (
 	"time"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	"github.com/onsi/gomega/types"
 	kubectl "github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	elementalv1 "github.com/rancher/elemental-operator/api/v1beta1"
 	upgradev1 "github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1"
 
 	"github.com/rancher/elemental-operator/tests/catalog"
 )
 
-const discoveryPluginImage = "quay.io/costoolkit/upgradechannel-discovery:v0.4.0"
+const discoveryPluginImage = `"quay.io/costoolkit/upgradechannel-discovery:v0.4.0"`
 
 func getPlan(s string) (up *upgradev1.Plan, err error) {
 	up = &upgradev1.Plan{}
 	err = kubectl.GetObject(s, cattleSystemNamespace, "plan", up)
 	return
 }
 
-func waitTestChannelPopulate(k *kubectl.Kubectl, mr *catalog.ManagedOSVersionChannel, name string, image, version string) {
+func waitTestChannelPopulate(k *kubectl.Kubectl, mr *elementalv1.ManagedOSVersionChannel, name string, image, version string) {
 	EventuallyWithOffset(1, func() error {
-		return k.ApplyYAML(fleetNamespace, name, mr)
+		return k.ApplyJSON("", name, mr)
```

Comment:
You can also use controller runtime client to create/update/delete object as we did here https://github.com/rancher/elemental-operator/blob/main/tests/e2e/machineregistration_test.go#L95

Reply:
Indeed, let me give it a try. I guess not all kubectl calls of this test can be replaced though.

Reply:
@alexander-demicev see my last commit, I used the controller runtime to create the CRDs instead of the kubectl wrapper with JSON inputs, however I could not manage to use the controller runtime to fetch CRDs that are not defined by this controller API, which is problematic for this tests as it operates with CRDS from `github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1`. For instance if trying to adapt https://github.com/rancher/elemental-operator/blob/3d0af6e62051a77ac557f921bad1793aa8fc8e46/tests/e2e/upgrades_test.go#L37-L41
I get a `no kind is registered for the type v1.Plan in scheme` error, which I assume is caused because `system-upgrade-controller` is not based on controller runtime. Do you know if this is something we can easily circumvent? Otherwise I am more leaning towards dropping my latest commit and do not use the controller runtime client for this test. I dislike using two different k8s clients on the same piece of code, specially for tests it turns to be confusing.

Reply:
I finally dropped the last commit and for now I did not use the client from controller runtime. As I could not figure out a proper way to use it for all cases required here in this test.